vendor/github.com/moby/buildkit/util/contentutil/copy.go

```diff
@@ -75,7 +76,7 @@ func CopyChain(ctx context.Context, ingester content.Ingester, provider content.
 		}
 	})
 	handlers := []images.Handler{
-		images.ChildrenHandler(provider),
+		annotateDistributionSourceHandler(images.ChildrenHandler(provider), desc.Annotations),
```

Comment:
Maybe the annotated descriptor should be already provided by the `content.Provider` implementation?